packages/thirdweb/src/react/web/ui/prebuilt/NFTProvider/useNFTMedia.ts

```diff
@@ -0,0 +1,100 @@
+import { useQuery } from "@tanstack/react-query";
+import type { Chain } from "../../../../../chains/types.js";
+import type { ThirdwebClient } from "../../../../../client/client.js";
+import { getContract } from "../../../../../contract/contract.js";
+import { punkImageSvg } from "../../../../../extensions/cryptopunks/__generated__/CryptoPunks/read/punkImageSvg.js";
+import { tokenURI } from "../../../../../extensions/erc721/__generated__/IERC721A/read/tokenURI.js";
+import { uri } from "../../../../../extensions/erc1155/__generated__/IERC1155/read/uri.js";
+import type { BaseTransactionOptions } from "../../../../../transaction/types.js";
+
+export function useNFTMedia(props: {
+  contractAddress?: string;
+  chain?: Chain;
+  client?: ThirdwebClient;
+  tokenId?: bigint;
+  disabled?: boolean;
+  overrideMediaField?: string;
+}) {
+  const {
+    contractAddress,
+    chain,
+    client,
+    tokenId,
+    disabled,
+    overrideMediaField,
+  } = props;
+  return useQuery({
+    queryKey: ["nft-media", chain?.id, contractAddress, tokenId?.toString()],
+    queryFn: async (): Promise<string | undefined> => {
+      if (!contractAddress) {
+        throw new Error("Error from useNFTMedia: Missing contract address");
+      }
+      if (!chain) {
+        throw new Error("Error from useNFTMedia: Missing chain");
+      }
+      if (!tokenId) {
+        throw new Error("Error from useNFTMedia: Missing tokenId");
+      }
+      if (!client) {
+        throw new Error("Error from useNFTMedia: Missing tokenId");
+      }
+      const contract = getContract({
+        address: contractAddress,
+        chain,
+        client,
+      });
+      return getNFTMedia({ contract, tokenId, client, overrideMediaField });
+    },
+    enabled: !disabled && !!contractAddress && !!chain && !!client && !!tokenId,
+  });
+}
+
+/**
+ * @internal
+ */
+export async function getNFTMedia(
+  options: BaseTransactionOptions<{
+    tokenId: bigint;
+    client: ThirdwebClient;
+    overrideMediaField?: string;
+  }>,
+): Promise<string> {
+  const { contract, tokenId, client, overrideMediaField } = options;
+  /**
+   * No need to check whether a token is ERC721 or 1155. Because there are
+   * other standards of NFT beside 721 and 1155 (404, onchain CryptoPunks etc.).
+   * So we run all possible uri-fetching methods at once and use the one that's available
+   */
+  const [_tokenURI, _uri, imageBase64, { fetchTokenMetadata }] =
+    await Promise.all([
+      tokenURI({ contract, tokenId }).catch(() => ""),
+      uri({ contract, tokenId }).catch(() => ""),
+      punkImageSvg({ contract, index: Number(tokenId) }).catch(() => ""),
+      import("../../../../../utils/nft/fetchTokenMetadata.js"),
+    ]);
```

Comment:
I don't think we should support anything no ERC721/ERC1155 standard here?

Reply:
where do u stop with that basically :) there are 1000+ "nft like" things onchain that all have custom handling for this, I don't think we can handle that here